services/task-svc/internal/bed/bed.go

```diff
@@ -97,15 +72,20 @@ func (ServiceServer) GetBed(ctx context.Context, req *pb.GetBedRequest) (*pb.Get
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	bed := models.Bed{ID: id}
-	if err := db.Where("organization_id = ?", organizationID).First(&bed).Error; err != nil {
+	bed, err := bedRepo.GetBedById(id)
+	if err != nil {
 		if hwgorm.IsOurFault(err) {
 			return nil, status.Error(codes.Internal, err.Error())
 		} else {
 			return nil, status.Error(codes.InvalidArgument, "id not found")
 		}
 	}
 
+	if bed.OrganizationID != organizationID {
+		// exceptionally rare to happen, as the uuidv4 space is huge
+		return nil, status.Error(codes.PermissionDenied, "bed does not belong to your organization")
+	}
```

Comment:
The organization check implies the necessarily of a `GetBedByIdForOrganization(id, organizationID)` method